manifests/00-custom-resource-definition.yaml

```diff
@@ -481,8 +128,6 @@ spec:
                     networking, and is not explicitly published. The user must manually
                     publish the ingress controller."
                   type: string
-              required:
-              - type
```

Comment:
this looks suspicious

Reply:
I added `+optional` when I added `+unionDiscriminator` in https://github.com/openshift/api/commit/2024105dd4c3cc30148d4127545eec7b21a53b65#diff-85d84174432a6d2df99d107129e012f8R246, but maybe I shouldn't have.  https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190325-unions.md#go-tags says, "`// +unionDiscriminator` before a field means that this field is the discriminator for the union. Only one field per structure can have this prefix. This field HAS TO be a string, and CAN be optional."

Reply:
@Miciah what do you propose we do here?

Reply:
I do not know of a reason why this field should be marked as optional (it came up during review: https://github.com/openshift/api/pull/312#discussion_r293022546, but I do not see a reason given why it should be optional).  `spec.endpointPublishingStrategy` is optional, so I believe `type` does not need to be so.  If it is not too late to do so, I would like to remove `+optional`.  The question that was raised in the discussion about the `+nullable` markers applies here:  Since the old generated CRD does not reflect the markers, is it safe to remove them?

Reply:
Like with nullable, how about open a PR and let's get the discussion centralized?

Reply:
https://github.com/openshift/api/pull/440

Reply:
Fixed